[main] Update dotnet/runtime

lewing opened the pull request:
Fix the 9.0.100 band in the emsdk transport package name

lewing committed:
[main] Update dotnet/runtime

mmitche commented on eng/Versions.props:
```diff
@@ -35,31 +35,31 @@
     <!-- Dependencies from https://github.com/dotnet/runtime -->
-    <MicrosoftNETCoreAppRefPackageVersion>8.0.0-rc.1.23421.3</MicrosoftNETCoreAppRefPackageVersion>
+    <MicrosoftNETCoreAppRefPackageVersion>9.0.0-alpha.1.23430.1</MicrosoftNETCoreAppRefPackageVersion>
     <VSRedistCommonNetCoreSharedFrameworkx6480PackageVersion>8.0.0-rc.1.23421.3</VSRedistCommonNetCoreSharedFrameworkx6480PackageVersion>
```
This will need to be updated.

mmitche replied:
(including name)

lewing committed:
Fix SharedFramework too

lewing commented:
Installer needs a bunch of updates as well once this lands

jeffschwMSFT commented:
who is on point to either identifying the individual failures and farming out the work and/or handling them?

lewing committed:
Update the variable used in global.json too

mmitche commented:
@jeffschwMSFT There shouldn't be too much. This was the commit to switch to 8: https://github.com/dotnet/sdk/commit/8119940c5bc28877b720e069c84995cdd1ba220e

lewing commented:
opened https://github.com/dotnet/arcade/pull/14002 for future me

lewing commented:
Looks like it is failing on rid resolution now @elinor-fung can you take a look?

elinor-fung commented:
The built tests are configured to run on a runtime from before https://github.com/dotnet/docs/issues/36466 for some reason.

If I build `main`, the runtimeconfig for the Microsoft.NET.Build.Tests has the runtime version corresponding to Version.props/Version.Details.xml:
```
  "runtimeOptions": {
    "tfm": "net8.0",
    "rollForward": "Major",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "8.0.0-rc.1.23421.3"
    },
```
But from this branch, it has:
```
  "runtimeOptions": {
    "tfm": "net8.0",
    "rollForward": "Major",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "8.0.0-preview.7.23375.6"
    },
```

I'm not sure why yet - cc @dsplaisted in case it is something obvious to you

elinor-fung commented:
This is what does the updating to the runtime version that gets used by tests:
https://github.com/dotnet/sdk/blob/aabbd9763acc35126db30445e0652e7dd821293f/Directory.Build.targets#L46-L54

But with this PR, `$(MicrosoftNETCoreAppRefPackageVersion.StartsWith('$(_TargetFrameworkVersionWithoutV)'))` is false:
- `MicrosoftNETCoreAppRefPackageVersion` is `9.0.0-alpha.1.23430.1`
- `_TargetFrameworkVersionWithoutV` is `8.0`

elinor-fung commented:
@dsplaisted / @marcpopMSFT how does SDK usually handle updating major versions and targeting the consumed runtime version? 

The retargeting for ASP.NET references has a similar condition:
https://github.com/dotnet/sdk/blob/aabbd9763acc35126db30445e0652e7dd821293f/Directory.Build.targets#L56-L63

Commit:
Update global.json

Commit:
Update blazor web tag

dsplaisted commented:
Now that the dependency flow is bringing in .NET 9 instead of .NET 8, the build doesn't have a version of .NET 8 to update the targeting to.  I think the thing to do here is to update the stage 0 SDK to one that has a recent enough version of the runtime.

FYI for those not aware, here is where we are trying to keep track of the changes needed each time we update to a new TargetFramework: https://github.com/dotnet/sdk/blob/main/documentation/general/UpdateToNewTargetFramework.md  Feel free to update it as appropriate.

Commit:
Update blazor web tag for osx and linux

lewing committed:
Update global.json

lewing committed:
Bump runtime dependencies

lewing commented:
@dotnet/domestic-cat please take a look

ViktorHofer commented:
@lewing what do you want me to review? I only see dependency and template updates.

agocke committed:
Bump tools packages to net8

agocke committed:
Suppress TFM build warnings temporarily

agocke commented:
There's a remaining build failure that is about differing versions of System.Reflection.Metadata. Not sure how this is happening.

lewing committed:
Update to more recent runtime build

lewing committed:
Update Directory.Build.props

lewing committed:
revert net9.0 target

nagilson committed:
Bump XUnitRunner TFM

The task target framework is grossly hardcoded
So when the helix project TFM was bumped the xunitrunner couldn't find it.
Hopefully this fixes some of the problems.

marcpopMSFT commented:
the helixtask overrides were unrelated to the SRM issue. I forget why I had to override them. You can try removing the override but I'd probably do that separately (if possible) to avoid adding to this complexity.

ViktorHofer commented on Directory.Build.props:
```diff
@@ -12,11 +12,13 @@
     <IncludeSymbols>true</IncludeSymbols>
     <IsShippingPackage>false</IsShippingPackage>
     <SdkTargetFramework>net8.0</SdkTargetFramework>
+    <!-- Temporarily disable build warnings while upgrading to the net9.0 TFM -->
+    <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings>
```
Why is that needed? That switch isn't intended to be set by our own repositories. It's a last effort escape switch.

marcpopMSFT replied:
The build was throwing a massive number of errors about incompatible runtime targeting. In prior years, we have not had to retarget the SDK before merging in a new runtime so this seems to be a new error:

marcpopMSFT replied:
So adding that flag was to suppress the error so we could get runtime merged and then retarget later (last year, retargetting happened ~3 months after rebranding)

ViktorHofer replied:
dotnet/runtime retargeting happen soon after rebranding this year while that happened around December last year. That's probably the reason why you weren't impacted by this.

IMO it makes sense to do the retargeting here as part of consuming the new runtime. IIRC that's what we did in the previous releases, before .NET 7. But if this switch is just here for a few weeks, then I'm fine with that.

marcpopMSFT replied:
ahh, I didn't realize runtime was that late last time around. That explains it. I'd like to take this whatever the shortest path to fixing the failures is

marcpopMSFT commented:
Latest errors appear to be caused by transitive pinning. I've pinged Jeff to see if he has a suggestion before I turn it off in main for now.

marcpopMSFT committed:
Disable transitive pinning as there are conflicts in SRM which we can't update yet. This should be temporary to unblock the build.

marcpopMSFT committed:
Merge remote-tracking branch 'origin/main' into update-emsdk-manifest

marcpopMSFT committed:
Reenable transitive pinning and apply a fix to the package that was causing prior errors

marcpopMSFT committed:
Switch a test to be self-contained. Not sure why this particular test was failing otherwise as it seemed to be trying to roll-forward across major.

marcpopMSFT committed:
Ran update-test-baselines.ps1

marcpopMSFT commented:
I pushed a fix to one of the SDK tests, I ran update baseline, and I fixed transitive pinning. Now we need @dotnet/sdk-container-builds-maintainers to investigate their test failures as those were the only ones left I hadn't gotten to I think.

baronfel commented:
@marcpopMSFT I took a brief look at the containers tests - they are building net8.0 apps targeting net8.0 runtimes (on purpose), but for some reason the containerized apps are trying to load System.Runtime 9.0.0.0. 

```shell
app@54c31fddd81f:/app$ cat CreateNewImageTest.runtimeconfig.json
{
  "runtimeOptions": {
    "tfm": "net8.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "8.0.0-preview.3.23174.8"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false,
      "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization": false
    }
  }
}app@54c31fddd81f:/app$ cat CreateNewImageTest.deps.json
{
  "runtimeTarget": {
    "name": ".NETCoreApp,Version=v8.0/linux-x64",
    "signature": ""
  },
  "compilationOptions": {},
  "targets": {
    ".NETCoreApp,Version=v8.0": {},
    ".NETCoreApp,Version=v8.0/linux-x64": {
      "CreateNewImageTest/1.0.0": {
        "dependencies": {
          "Microsoft.NET.Build.Containers": "9.0.100-dev"
        },
        "runtime": {
          "CreateNewImageTest.dll": {}
        }
      },
      "Microsoft.NET.Build.Containers/9.0.100-dev": {}
    }
  },
  "libraries": {
    "CreateNewImageTest/1.0.0": {
      "type": "project",
      "serviceable": false,
      "sha512": ""
    },
    "Microsoft.NET.Build.Containers/9.0.100-dev": {
      "type": "package",
      "serviceable": true,
      "sha512": "sha512-BxHLUP94QlJT/E3Cm5G9a+sQfg0SmWaUTdnXPI5QXS6H8axKSjO8HmIWdcc+tycpA1jgC4b6+Flvke190tJqKQ==",
      "path": "microsoft.net.build.containers/9.0.100-dev",
      "hashPath": "microsoft.net.build.containers.9.0.100-dev.nupkg.sha512"
    }
  }
}
```

The deps.json and runtimeconfig.json here both show targeting .net 8, so I'm not sure where this load of the 9.0 dll is coming from. COREHOST_TRACE was inconclusive on first load as well.

ViktorHofer commented on src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/Microsoft.NET.Build.Tasks.UnitTests.csproj:
```diff
@@ -20,9 +20,13 @@
     <PackageReference Include="Microsoft.Build.Utilities.Core" />
     <PackageReference Include="Microsoft.Extensions.DependencyModel" />
     <PackageReference Include="NuGet.ProjectModel" />
+    <PackageReference Include="System.Security.Cryptography.X509Certificates" />
```
You shouldn't need a package reference to X509Certificates. The package hasn't shipped since 2017 and will soon be marked as deprecated. You don't need the PackageReference on .NET Framework either as the X509Certificates types are inbox.

marcpopMSFT commented:
> @marcpopMSFT I took a brief look at the containers tests - they are building net8.0 apps targeting net8.0 runtimes (on purpose), but for some reason the containerized apps are trying to load System.Runtime 9.0.0.0.
@vitek-karas @agocke either of you know why the container tests appear to be looking for the 9.0 runtime even though that's not listed in the runtimeconfig.json? Starting with ya'll as a shot in the dark.

vitek-karas commented:
>  either of you know why the container tests appear to be looking for the 9.0 runtime even though that's not listed in the runtimeconfig.json? Starting with ya'll as a shot in the dark.

I assume this is about a failure like this:
```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

That would mean the assembly which you're trying to run was compiled referencing `System.Runtime, Version=9.0.0.0`. The content of `.runtimeconfig.json` only tells the runtime what framework to load. So if it was told to load net8, but then the code you're trying to run references 9.0 assemblies - that won't work.

It would work the other way round - the runtime will happily resolve a reference for `System.Runtime, Version=8.0.0.0` if it has 9.0 loaded. But it will not allow a downgrade.

baronfel commented:
@vitek-karas good idea. I took the a failing test locally and investigated. What's happening is that the _app_ is correctly targeting `net8.0`, but the Microsoft.NETCore.App KnownFrameworkReference from the BundledVersions.props is pulling in the 9.0-versions which are incorrectly categorized as `net8.0`:

![image](https://github.com/dotnet/sdk/assets/573979/8d5e4d49-298f-4e15-8d00-cf33033b37b8)

That BundledVersions props file is generated from dotnet/installer - so I'm not sure we have the correct knobs to make it work at this point. It feels wrong that the version slipped to point to 9 - should the 8.x KnownFrameworkReference be pinned to an 8.0 reference before anything else happens?

marcpopMSFT commented:
@rainersigwald @dsplaisted I've been investigating a set of tests that are failing to load System.Text.Json. I've tried adding direct references to the that dll in the build tasks csproj (both the 9.0 and 8.0 targeted reference) but neither has worked. Other suggestions?

```
  C:\h\w\960D085D\p\d\sdk\9.0.100-ci\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(783,5): error MSB4018: The "GenerateClsidMap" task failed unexpectedly. [C:\h\w\960D085D\t\dotnetSdkTests\nojy2rp0.mjd\It_embeds_sin---38A77651\ComServerWithTypeLibs.csproj]
      C:\h\w\960D085D\p\d\sdk\9.0.100-ci\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(783,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified. [C:\h\w\960D085D\t\dotnetSdkTests\nojy2rp0.mjd\It_embeds_sin---38A77651\ComServerWithTypeLibs.csproj]
      ```

baronfel commented:
> @baronfel BundledVersions.props is generated in dotnet/installer, but we also do some processing on it in dotnet/sdk for the stage 2 layout of the SDK. So check to see whether the .NET 8 / 9 mismatch is present in the stage 0 bundled versions file. If not then there's an issue in the SDK repo (probably in the OverrideAndCreateBundledNETCoreAppPackageVersion task).

This was exactly right - stage0 looks correct:

```xml
<!-- .NET 8.0 -->
    <KnownFrameworkReference Include="Microsoft.NETCore.App"
                              TargetFramework="net8.0"
                              RuntimeFrameworkName="Microsoft.NETCore.App"
                              DefaultRuntimeFrameworkVersion="8.0.0-rc.1.23414.4"
                              LatestRuntimeFrameworkVersion="8.0.0-rc.1.23414.4"
                              TargetingPackName="Microsoft.NETCore.App.Ref"
                              TargetingPackVersion="8.0.0-rc.1.23414.4"
                              RuntimePackNamePatterns="Microsoft.NETCore.App.Runtime.**RID**"
                              RuntimePackRuntimeIdentifiers="linux-arm;linux-arm64;linux-musl-arm64;linux-musl-x64;linux-x64;osx-x64;rhel.6-x64;tizen.4.0.0-armel;tizen.5.0.0-armel;win-arm64;win-x64;win-x86;linux-musl-arm;osx-arm64;linux-s390x;linux-bionic-arm;linux-bionic-arm64;linux-bionic-x64;linux-bionic-x86;linux-ppc64le"
                              />
```

marcpopMSFT committed:
Target a test to be self-contained
hardcode system text json to the 8.0 RC1 version for the task assembly

marcpopMSFT committed:
Merge remote-tracking branch 'origin/main' into update-emsdk-manifest

marcpopMSFT commented:
> @rainersigwald @dsplaisted I've been investigating a set of tests that are failing to load System.Text.Json. I've tried adding direct references to the that dll in the build tasks csproj (both the 9.0 and 8.0 targeted reference) but neither has worked. Other suggestions?

FWIW, I tried hardcoding to system.text.json a second time and this time the tests passed. Not sure what I did the prior time (perhaps I didn't fully rebuild). Not sure if this is the right fix but we'll see test results

baronfel committed:
don't overwrite bundledversions unless the SemVer major values match

baronfel commented:
After talking with @dsplaisted and the team, we decided to try a change to the way the SDK lays out the BundledVersion props for the 'redist' SDK - if the major version of the Stage0 SDK and the NETCore.App Runtime Pack dependency of the SDK itself differs, we just won't apply the SDK's change to the bundled versions. This will let us continue to support the previous TFM until the time that installer updates it's BundledVersions.props to include net9.0-targeting packs.

I made this change and commented it, and it does seem to unblock some number of the containers tests.

baronfel commented on src/Layout/toolset-tasks/OverrideAndCreateBundledNETCoreAppPackageVersion.cs:
```diff
@@ -56,9 +57,24 @@ public override bool Execute()
-                propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value;
-            propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value = microsoftNETCoreAppRefPackageVersion;
+
+            var originalBundledNETCoreAppPackageVersion = propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value;
+            var parsedOriginalBundledPackageVersion = SemanticVersion.Parse(originalBundledNETCoreAppPackageVersion);
+            var parsedMicrosoftNETCoreAppRefPackageVersion =
+                SemanticVersion.Parse(microsoftNETCoreAppRefPackageVersion);
+
+            // In the case where we have a new major version, it'll come in first through the dotnet/runtime flow of the
+            // SDK's own package references. The Stage0 SDK's bundled version props file will still be on the older major version
+            // (and the older TFM that goes along with that). If we just replaced the bundled version with the new major version,
+            // apps that target the 'older' TFM would fail to build. So we need to keep the bundled version from the existing
+            // bundledversions.props in this one specific case.
+
+            var newBundledPackageVersion =
+                parsedOriginalBundledPackageVersion.Major == parsedMicrosoftNETCoreAppRefPackageVersion.Major
+                ? microsoftNETCoreAppRefPackageVersion
+                : originalBundledNETCoreAppPackageVersion;
+
+            propertyGroup.Element(ns + "BundledNETCoreAppPackageVersion").Value = newBundledPackageVersion;
 
```
@dsplaisted would you mind double-checking this and seeing if it matched what you were talking about?

dsplaisted replied:
Yes, this matches what I expected.  It might be better to use `NuGetVersion` instead of `SemanticVersion`.  I'm not too sure of the differences between them, but the code I've seen has used `NuGetVersion`.

marcpopMSFT committed:
Add an additional dependency that's flagged in the tests.

marcpopMSFT committed:
Remove the cryptography reference per PR feedback

elinor-fung commented:
> Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAnAotApp.NativeAotLib_errors_out_when_eventpipe_is_enabled

`Microsoft.DotNet.ILCompiler` being used is now `8.0.0-rc.1.23414.4` - which is from before https://github.com/dotnet/runtime/pull/90811 (main is using `8.0.0-rc.1.23421.3`). Maybe related to the BundledVersions changes in this PR?

Side note: not the issue being hit in this PR, but that error is also being downgraded to a warning (https://github.com/dotnet/runtime/pull/91715), so the test will also need to be updated when that change flows over.

lewing committed:
Bump runtime build

lewing commented:
I've updated the runtime flow to the most recent build. The wasm build is odd @maraf and @radical can you please take a look.

elinor-fung commented:
> C:\h\w\B1C3097B\t\dotnetSdkTests\ij4x4eyn.d0b\.nuget\packages\microsoft.netcore.app.runtime.mono.browser-wasm\8.0.0-rc.1.23414.4\runtimes\browser-wasm\native\libmono-wasm-simd.a: No such file or directory

Maybe the wasm error is the same as root cause as the native aot one - seems like the test environment gets constructed such that it is using older packages. As with `Microsoft.DotNet.ILCompiler`, the version of `Microsoft.NETCore.App.Runtime.Mono.browser-wasm` being used is an older version - `8.0.0-rc.1.23414.4` - which is before `libmono-wasm-simd.a` was added.

radical commented:
> C:\h\w\B1C3097B\t\dotnetSdkTests\ij4x4eyn.d0b.nuget\packages\microsoft.netcore.app.runtime.mono.browser-wasm\8.0.0-rc.1.23414.4\runtimes\browser-wasm\native\libmono-wasm-simd.a: No such file or directory

This is because 9.0 targets are being used with a 8.0 runtime pack. I'm trying to reproduce this locally now.

radical committed:
Fix blazor tests built with workloads

The tests fail with:
```
emcc : error : /private/tmp/helix/working/A19408C3/w/B54A09BA/e/testExecutionDirectory/.nuget/packages/microsoft.netcore.app.runtime.mono.browser-wasm/8.0.0-rc.1.23414.4/runtimes/browser-wasm/native/libmono-wasm-simd.a: No such file or directory
```

.. because the build is using 9.0 targets, but 8.0 runtime pack.

This commit works around that issue by overriding the runtime pack path
to point to the correct one. `WasmOverridePacks.targets` is taken from
`Wasm.Build.Tests` in `dotnet/runtime`.

This workaround can be removed once we have a proper 9.0 workload.

radical committed:
Use WasmOverride.targets only for net8.0 wasm projects

lewing commented:
Looks like we are down to NativeAOT publish failures now?

marek-safar commented:
@sbomer @LakshanF could you help with resolving tests failures in AotStaticLibraryPublishWithEventPipe ?

marek-safar committed:
Merge branch 'main' into update-emsdk-manifest

LakshanF committed:
Fix native aot lib test scenario

marek-safar commented:
@dotnet/source-build-internal could you please advise how to track where these prebuilts are coming from and how to version `baseline-comparison.xml` in 8.0 arcade for 9.0 main?

mthalman committed:
Update versions to official RC1

MichaelSimons commented on eng/Version.Details.xml:
```diff
@@ -14,46 +14,46 @@
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.DotNetHostResolver" Version="8.0.0-rc.1.23421.3">
+    <Dependency Name="Microsoft.NETCore.DotNetHostResolver" Version="9.0.0-alpha.1.23455.1">
       <Uri>https://github.com/dotnet/runtime</Uri>
```
Is this intentionally incoherent with the other runtime dependencies?  They are on version 9.0.0-alpha.1.23464.19 and sha a6c64c3c26417f579022e20cbcee992b4814ed2c. This is the source of the following two source-build prebuilts that are being reported:

```
    <Usage Id="Microsoft.NETCore.DotNetHostResolver" Version="9.0.0-alpha.1.23455.1" File="src/artifacts/obj/Microsoft.DotNet.MSBuildSdkResolver/project.assets.json" IsDirectDependency="true" />
    <Usage Id="Microsoft.NETCore.DotNetHostResolver" Version="9.0.0-alpha.1.23455.1" File="src/artifacts/obj/Microsoft.DotNet.SdkResolver/project.assets.json" IsDirectDependency="true" />
```

elinor-fung replied:
We actually removed `Microsoft.NETCore.DotNetHostResolver` in runtime. It looks like this repo is using it to grab `hostfxr` for Framework builds. I think it can use `Microsoft.NETCore.App.Runtime.*` packages - I'll take a look.

MichaelSimons commented on src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj:
```diff
@@ -52,8 +52,10 @@
     <PackageReference Include="Microsoft.Build.Utilities.Core" ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" />
-    <PackageReference Include="Microsoft.NET.HostModel" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" VersionOverride="8.0.0-rc.1.23419.4"/>
```
@marcpopMSFT - I see you hard coded the version in https://github.com/dotnet/sdk/commit/3cd48fd80f26c5a23cc2f0c69cba738764b149c0.  I am assuming this is a temporary workaround of sorts.  If that is the case, then an exception should be added to the [prebuilt baseline](https://github.com/dotnet/sdk/blob/main/eng/SourceBuildPrebuiltBaseline.xml)

```
    <!-- Add comment describing the temporary nature of these and link a tracking issue to remove. -->
    <UsagePattern IdentityGlob="Microsoft.NET.HostModel/8.0.0-rc.1.23419.4" />
    <UsagePattern IdentityGlob="System.Text.Json/8.0.0-rc.1.23419.4" />
    <UsagePattern IdentityGlob="Microsoft.Bcl.AsyncInterfaces/8.0.0-rc.1.23419.4" />
```

marcpopMSFT replied:
correct, we were getting task exceptions because the 8.0 system.text.json wouldn't load so I had to add a hardcoded package reference in the task assembly and then add 3 more dependencies of that package.

MichaelSimons commented:
> @dotnet/source-build-internal could you please advise how to track where these prebuilts are coming from and how to version `baseline-comparison.xml` in 8.0 arcade for 9.0 main?

I took a quick look.  Here is what I did:

1. Reviewed the prebuilt [baseline-comparison.xml.](https://artprodcus3.artifacts.visualstudio.com/A6fcc92e5-73a7-4f88-8d13-d9045b45fb27/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy1wdWJsaWMvcHJvamVjdElkL2NiYjE4MjYxLWM0OGYtNGFiYi04NjUxLThjZGNiNTQ3NDY0OS9idWlsZElkLzQxMDY0Mi9hcnRpZmFjdE5hbWUvQnVpbGRMb2dzX1NvdXJjZUJ1aWxkX01hbmFnZWRfQXR0ZW1wdDE1/content?format=file&subPath=%2Fartifacts%2Fsource-build%2Fself%2Fprebuilt-report%2Fbaseline-comparison.xml)
2. Noted two classes of prebuilts as they come from two different project groupings.
3. First grouping is from src/artifacts/obj/Release/Sdks/Microsoft.NET.Sdk/tools/project.assets.json which when looking at the changes led me to [this conclusion](https://github.com/dotnet/sdk/pull/35048/files#r1330201831).
4. Second grouping is from src/artifacts/obj/Microsoft.DotNet.MSBuildSdkResolver/project.assets.json and src/artifacts/obj/Microsoft.DotNet.SdkResolver/project.assets.json which led me to review the version of the referenced prebuilts in the versions.props and version.details.xml and noticed [this incoherency](https://github.com/dotnet/sdk/pull/35048/files#r1330167908).

elinor-fung committed:
Remove Microsoft.NETCore.DotNetHostResolver dependency

elinor-fung commented:
Second grouping from Microsoft.DotNet.MSBuildSdkResolver / Microsoft.DotNet.SdkResolver is resolved.

https://github.com/dotnet/sdk/pull/35048#discussion_r1330201831 is all that is left.

lewing commented:
@marcpopMSFT ping

marcpopMSFT committed:
Update SourceBuildPrebuiltBaseline.xml for 8.0 msbuild dependencies
